api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -0,0 +1,251 @@
+            }
+        }
+
+        for (ReportDesign reportDesign : reportDesigns) {
+            Context.getService(ReportService.class).saveReportDesign(reportDesign);
+        }
+
+    }
+
+    public static  ReportDefinition constructReportDefinition(ReportDescriptor reportDescriptor) {
+
+        ReportDefinition rd = new ReportDefinition();
+        // TODO do we need to handle translations here?
+        rd.setName(reportDescriptor.getName());
+        rd.setDescription(reportDescriptor.getDescription());
+        rd.setUuid(reportDescriptor.getUuid());
+
+        List<Parameter> parameters = constructParameters(reportDescriptor.getParameters());
+        Map<String, Object> mappings = constructMappings(parameters);
+        rd.setParameters(parameters);
+
+        if (reportDescriptor.getDatasets() != null) {
+            for (DataSetDescriptor dataSetDescriptor : reportDescriptor.getDatasets()) {
+                DataSetDefinition dsd = constructDataSetDefinition(dataSetDescriptor, parameters);
+                // TODO do we need to handle DSD name and description?
+                if (dsd != null) {
+                    rd.addDataSetDefinition(dataSetDescriptor.getKey(), dsd, mappings);
+                }
+            }
+        }
+        return rd;
+    }
+
+    public static List<Parameter> constructParameters(List<ParameterDescriptor> parameterDescriptors) {
+        List<Parameter> parameters = new ArrayList<Parameter>();
+        if (parameterDescriptors != null) {
+            for (ParameterDescriptor parameterDescriptor : parameterDescriptors) {
+                try {
+                    // TODO how else might we want to configure parameters? ie other "types" besides classes?
+                    parameters.add(new Parameter(parameterDescriptor.getKey(), parameterDescriptor.getLabel(), Class.forName(parameterDescriptor.getType())));
+                } catch (Exception e) {
+                    log.error("Unable to configure parameter " + parameterDescriptor.getKey(), e);
+                }
+            }
+        }
+        return parameters;
+    }
+
+
+    public static Map<String, Object> constructMappings(List<Parameter> parameters) {
+        // TODO: Mike is the correct and/or necessary?
+        Map<String,Object> mappings = new HashMap<String, Object>();
+        for (Parameter parameter : parameters) {
+            mappings.put(parameter.getName(), "${" + parameter.getName() + "}");
+        }
+        return mappings;
+    }
+
+    public static DataSetDefinition constructDataSetDefinition(DataSetDescriptor dataSetDescriptor, List<Parameter> parameters) {
+
+        DataSetDefinition dsd = null;
+
+        if ("sql".equalsIgnoreCase(dataSetDescriptor.getType())) {
+            dsd = constructSQLFileDataSetDefinition(dataSetDescriptor);
+        }
+        else {
+            log.error("Unsupported data set descriptor type: " + dataSetDescriptor.getType());
```

Comment:
Why log.error?  You don't think we should fail hard here with a runtime exception?

Reply:
My main thought is that we were trying to get away from having an error like this take down all of the PIH EMR, right?  If we through a Runtime, Reporting won't start and therefore most modules won't start.  

Especially now since this can be updated in the config without doing a code update (and even more so if we start allowing hot reloads).  Thoughts on that?

Reply:
If we start doing hot reloads, then yes.  But otherwise I do think that we want to fail, just like a mis-configured piece of metadata for Iniz should fail.